### ISSUE TITLE
Integrate real data from the OpenSpending API

### ISSUE THREAD

**Issue by pwalsh** [openspending/subsidystories.eu#2]
# Description

We want to populate the mock data with real data calls from the OpenSpending API.

# Tasks

- [ ] @akariv to define.

**Comment by akariv** [comment]
@gocemitevski: So data is good enough to start working with it.

Here's a spec of what needs to be done:
- [x] Modify the global page so that each click on a country would open the country page
- [x] Pass the country code via the URL (e.g. query parameter)
- [x] Same URLs should be used in the right side navigation bar
- [x] The country page should use the query parameter to fetch data for that country code:
   - [x] The sum of all subsidies if fetched using this API call: 
          `http://next.openspending.org/api/3/cubes/<dataset-name>/aggregate`
          With the following parameters:
          - `aggregates` = `total_amount.sum`
          - `cut` = `administrative_classification_2.beneficiary_country_code:"<country-code>"`
          - `drilldown` = `activity_funding_period_number.funding_period_number`
          See an example request [here](http://next.openspending.org/api/3/cubes/6018ab87076187018fc29c94a68a3cd2:eu-esif-funds-beneficiaries-2007-2020/aggregate?aggregates=total_amount.sum&cut=administrative_classification_2.beneficiary_country_code:%22PL%22&drilldown=activity_funding_period_number.funding_period_number)
   - [x] The top 10 beneficiaries are fetched using this API call:
          `http://next.openspending.org/api/3/cubes/<dataset-name>/aggregate`
          With the following parameters:
          - `aggregates` = `total_amount.sum`
          - `cut` = `administrative_classification_2.beneficiary_country_code:"<country-code>"`
          - `drilldown` = `recipient_2.beneficiary_name`
          - `pagesize` = `10`
          - `order` = `total_amount.sum:desc`
          See an example request [here](http://next.openspending.org/api/3/cubes/6018ab87076187018fc29c94a68a3cd2:eu-esif-funds-beneficiaries-2007-2020/aggregate?aggregates=total_amount.sum&cut=administrative_classification_2.beneficiary_country_code:%22PL%22&drilldown=recipient_2.beneficiary_name&pagesize=10&order=total_amount.sum:desc)
   - [x] Other visualisations will be embeds from OpenSpending's OS-Viewer, for example [this one](http://next.openspending.org/viewer/embed/treemap/6018ab87076187018fc29c94a68a3cd2:eu-esif-funds-beneficiaries-2007-2020?lang=en&measure=total_amount.sum&groups%5B%5D=recipient_2.beneficiary_name&filters%5Badministrative_classification_2.beneficiary_country_code%5D%5B%5D=PL&order=total_amount.sum%7Cdesc)
           Basically the URL format would be `http://next.openspending.org/viewer/embed/<embed-type>/<dataset-name>?<embed-params>`, where in the embed params you'd have to add the country code somewhere.
           The list of embeds should be part of the configuration (see below), with the only processing is the "country-code embedding" I just mentioned.
   - [x] you can use jQuery (or similar) to fetch the data
- [x] There should be some sort of configuration file which is separated from the code, containing:
  - [x] The name of the dataset (currently `6018ab87076187018fc29c94a68a3cd2:eu-esif-funds-beneficiaries-2007-2020`)
  - [x] the list of embed URLs to use in the main grid.

Regarding the currently existing pages:
- [x] global page (`index.html`) should remain static
- [x] country page (`country.html`) should be dynamic based on the info above
- [x] details page (`country-details.html`) is out of scope at the moment until it's better defined.

cc: @pwalsh @annafralberts @cyberbikepunk

**Comment by cyberbikepunk** [comment]
@gocemitevski Looking forward to having a nice front-end! We're actively working with @michaeljokf on putting some more category information in there. Let's see how the above goes and I'll ping you as soon as I have some more query possibilities for you.

**Comment by akariv** [comment]
@michaeljokf  and  @cyberbikepunk 
Just to keep the process straight, once we finish this issue (they way it's defined above ^), any change request should pass through an issue in this repo (naturally you can also change the code on your own and submit a pull request). 

The decision of which queries should appear on the country page is a product decision, please give it some holistic though before pushing any requests here 😄

**Comment by annafralberts** [comment]
Hey! @gocemitevski where are we at?

On thoughts for the design, the first idea:
1. Small version of the overall dataset with the following columns: Beneficiary Name, Project Name
Region, Fund, Start Date, Complete Date, Total Amount.


2. 
Standard view of the dataset:
Regional Division on the map, where you can zoom in on regions and how much money they have received compared to other regions (Jenkins thematic map).

Basic descriptives: 
Count: total number of projects, and total number of recipients.
Average amount received per beneficiary, average amount per project, median project. 
Distribution of the values (amounts) over the range. Y-axis, number of projects, X-axis, amount. 
Ten Highest Recipients 
Tree-Map
Time-line
Bar Chart comparing regions]

3. Documentation and suggestions for analysis. 

4. Policy resources. 

@seegerseeger suggestions and ideas how to display this?

**Comment by akariv** [comment]
Hey @annafralberts !

Currently @kravets-levko is implementing the website and @gocemitevski will take care of the design once the implementation is done.

As for the ideas you mentioned - these are actually not design decisions but rather product decisions. To avoid future frustrations, I suggest we follow this process:
- You and @seegerseeger should do the product thinking and decision making and update the mockup at os-data/eu-structural-funds#191 (which is currently our working implementation reference).
   We use the mockup as a tool to make sure that requirements are clear and that everyone is on the same page, so it doesn't have to be pretty - simply enough to make the point.
- Based on the requirements + mockup, I write a detailed implementation spec for each of the implementors (e.g. histograms are currently not possible in OpenSpending, and if it's a priority requirement we'll have to find a different solution to implement that).
- Implementors do their thing based on the spec
- Once things are stable enough, design work can start (but it won't change functionality - only flow,  interaction and appearance)
